web3-functions/pyth/index.ts

```diff
@@ -0,0 +1,338 @@
+    };
+  }
+  let pythConfig;
+
+  // Fetch Pyth config from gist
+  try {
+    pythConfig = await fetchPythConfigIfNecessary(storage, gistId);
+  } catch (err) {
+    const error = err;
+    return {
+      canExec: false,
+      message: `Error fetching gist: ${error}`
+    };
+  }
+  const debug = pythConfig.debug;
+  if (debug) {
+    console.debug(`pythConfig: ${JSON.stringify(pythConfig)}`);
+  }
+
+  const {
+    pythNetworkAddress,
+    priceServiceEndpoint,
+    validTimePeriodSeconds,
+    deviationThresholdBps,
+    priceIds,
+    composedDeviationThresholdBps,
+    composedPriceIds
+  } = pythConfig;
+  const pythContract = new Contract(
+    pythNetworkAddress,
+    PythAbi,
+    provider
+  );
+
+  const connection = new EvmPriceServiceConnection2(priceServiceEndpoint);
+  
+
+  // Add composed PriceFeeds to priceIds
+  const allComposedPriceIds: string[] = [];
+  const allPriceIds: string[] = []
+
+  if (priceIds) {
+    for (const priceId of priceIds) {
+      allPriceIds.push(addLeading0x(priceId));
+    }
+  }
+
+  if (composedPriceIds) {
+    for (const composedPriceId of composedPriceIds) {
+      for (const priceId of composedPriceId) {
+        allComposedPriceIds.push(addLeading0x(priceId));
+        if (!allPriceIds.includes(priceId)) {
+          allPriceIds.push(addLeading0x(priceId));
+        }
+      }
+    }
+  }
+
+  if (debug) {
+    console.debug(`fetching current prices for priceIds: ${allPriceIds}`);
+  }
+
+  const currentPrices = await getCurrentPrices(allPriceIds, connection, debug);
+  
+  if (currentPrices === void 0) {
+    return {
+      canExec: false,
+      message: `Error fetching latest priceFeeds for priceIds: ${allPriceIds}`
+    };
+  }
+  if (currentPrices.size != allPriceIds.length) {
+    const missingPriceIds = allPriceIds.filter((p) => !currentPrices.has(p));
+    console.error(
+      `Missing latest price feed info for ${JSON.stringify(missingPriceIds)}`
+    );
+    return { canExec: false, message: "Not all prices available" };
+  }
+  const lastPrices = await getLastPrices(allPriceIds, storage);
+  
+  if (debug) {
+    console.debug(
+      `
+        currentPrices: ${JSON.stringify([...currentPrices.entries()])}
+        lastPrices: ${JSON.stringify([...lastPrices.entries()])}
+      `
+    );
+  }
+
+  // Composed price feed needs update
+  const composedPriceFeedNeedsUpdate = (composedPriceIds: string[][]) => {
+    // Update all oracles if one of them needs update
+    let composedLastPrice = 0;
+    let composedCurrentPrice = 0;
+    let priceIsStale = false;
+    for (const priceId of composedPriceIds) {
+      // Update the pricefeed if we don't have the last price in storage
+      if (lastPrices.get(priceId) === void 0) {
+        console.log("Price needs to be loaded into storage")
+        return true;
+      }
+      
+      const currentPrice = currentPrices.get(priceId);
+      const lastPrice = lastPrices.get(priceId);
+
+      if (composedLastPrice === 0) {
```

Comment:
not that easy, here we always divide by the new price but it may happen that we need to multiply.

For instance imagine we have a EUROC/USD and a USD/EUR feed on Pyth --> we need to multiply them
Now if Pyth has feeds as EUROC/USD and a EUR/USD, the second ne need to divide the first one.